src/controller/current_loot.controller.ts

```diff
@@ -0,0 +1,254 @@
+import pkg, { create } from "lodash";
+const { difference } = pkg;
+import { CurrentLoot } from "../models/current_loot.model.js";
+import {
+  createCurrentLootPayload,
+  updateCurrentLootPayload,
+  CurrentLootServiceReturn,
+  findCurrentLootPayload,
+  deleteCurrentLootPayload,
+} from "./types/current_loot_types.js";
+import { Request, Response } from "express"; // Make sure you are importing from express
+
+///////Read or Pull all
+//need to add promises
+export const getAllCurrentLoot = async (
+  req: Request,
+  res: Response
+): Promise<CurrentLootServiceReturn> => {
+  try {
+    const getAllCurrentLoot = req.query;
+    console.log("Calling find all desirable loot");
+    const allCurrentLoot: CurrentLoot[] = await CurrentLoot.findAll({
+      where: getAllCurrentLoot,
+      order: [["id", "asc"]],
+    });
+    res.status(200).json({
+      status: "Sucesss",
+      data: allCurrentLoot,
+    });
+    return {
+      status: "Sucess",
+      data: allCurrentLoot,
+    };
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({
+      status: "Error",
+      data: error.message,
+    });
+    return {
+      status: "Error",
+      data: error.message,
+    };
+  }
+};
+
+export const getSingleCurrentLoot = async (
+  req: Request<{ id: number }, {}, {}, createCurrentLootPayload>, //You should build scalability with queries in the future, this is good for now though
+  res: Response
+): Promise<CurrentLootServiceReturn> => {
+  const findSingleLoot = req.query;
```

Comment:
Type :eyes:

Reply:
wait LMAO remove this

Reply:
wait hold up, this should be used in findSingleCurrentLoot query, don't delete but implement this in the SQL query below